app/controllers/concerns/cache_concern.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module CacheConcern
+  extend ActiveSupport::Concern
+
+  def render_with_cache(**options)
+    raise ArgumentError, 'only JSON render calls are supported' unless options[:json].present? || block_given?
+
+    key        = options.delete(:key) || request.url
+    expires_in = options.delete(:expires_in) || 3.minutes
+    body       = Rails.cache.read(key)
+
+    if body
+      render(options.except(:json, :serializer, :each_serializer, :adapter, :fields).merge(json: body))
+    else
+      options[:json] = yield if block_given?
```

Comment:
I was considering allowing calls like `render_with_cache json: -> { heavy_stuff_here }` initially but allocating Procs is slightly heavier than using yield, plus it's not that pretty either. However, supporting `render_with_cache json: :method_name_here` might be worth it..?

Reply:
I don't know Ruby enough to know exactly how this would work, but yeah, that might be worth it.